### ISSUE TITLE
CS1573 warning shouldn't appear when inheriting docs

### ISSUE THREAD

**Issue by AArnott** [dotnet/roslyn#40325]
The new `<inheritdoc cref />` support is *awesome*. I use it now to avoid a lot of redundancy in documentation across overloads. But I found that if I document one method then inherit that doc on another method that adds just one parameter, I get a CS1573 warning when I add documentation for just the one new parameter:

[Repro on sharplab](https://sharplab.io/#v2:EYLgHgbALANALiAlgGwD4AEBMBGAsAKHQGYACAQ2AGc4AnMgYzhKxIGESBvAknkgegEkAPJQCuAW3FkaATwB8AEQD2AU0olKS8SrgALRADsA5kL5jJ0+d14C+wgA7Sy4kgecqAvACJpR7F7kAMUQaanIaEz5HOnE5ax5icipaBiZ0KBJApSUACnRsAAZwvwBKAG4CeP5BIWjnV3dvXwKAgEEDEhUwFJIAFRl7FWLTOtiq22FDXRUaRDgAEyV6EnoaFQAzbyzc/IKSr344/F5mUgpqOkZmDO2c/sHigphmQuLscoIAXyA). Although sharplab doesn't show the intellisense that VS does, so here is the screenshot:

![image](https://user-images.githubusercontent.com/3548/70682842-5dd50800-1c5d-11ea-909b-8ee7b2d29a23.png)

The irony from the image above is that the popup when hovering over `arg1` *both* claims that there is no documentation for that parameter while also presenting the documentation (that was inherited from the other overload).

**Comment by AArnott** [comment]
@sharwell who added the inheritdoc feature recently.

**Comment by sharwell** [comment]
The compiler doesn't actually understand inheritdoc, which is an IDE feature. I would recommend inheriting the other way around. Also, consider disabling CS1573 and using a more configurable analyzer for documentation comments.

**Comment by AArnott** [comment]
Ya, the fact that it's not in the compiler itself is a bit unfortunate, since it means when I use `<inheritdoc/>` it will only provide documentation to 16.4+ users. But it's still great.

I can't inherit the other way around because the method with more parameters is conditionally defined while the method with fewer parameters is always defined.

I may disable CS1573, but is there another analyzer that will still help me make sure things are adequately documented?

**Comment by jinujoseph** [comment]
It needs a cross team work between compiler and IDE

**Comment by andre-ss6** [comment]
Does this means that `/// <inheritdoc />` is actually emitted in the resulting XML?

**Comment by AArnott** [comment]
Yes, @andre-ss6. The compiler does *not* generate xml with the docs actually copied -- it merely adds the reference. It's the language service that has to follow the refs to find the actual doc to present to the user in the editor.

**Comment by alexmg** [comment]
I have noticed the same issue with CS1573 being reported when there is a mix of `include file='foo.xml' path='doc/blah/*'` and direct parameter content.

**Comment by sharwell** [comment]
Moving this back to compilers. Their decision on whether CS1573 appears in this case is final in the sense that we can't override it from the IDE side.

**Comment by ZacharyPatten** [comment]
This is related to https://github.com/dotnet/roslyn/issues/42392. https://github.com/dotnet/roslyn/issues/42392 is having false positives on the base of the `inheritdoc` while https://github.com/dotnet/roslyn/issues/40325 is having false positives on the child of the `inheritdoc`.

Also, don't forget `CS1712`. Both `CS1573` & `CS1712` have the same issue. @AArnott can you add `CS1712` to the title of the issue? I doubt it is necessary to open a new issue just to document `CS1712` too.

### Examples
```cs
/// <summary>hello world</summary>
public static class CS1573_Issue
{
	/// <summary>hello world</summary>
	/// <param name="a">hello world</param>
	public static void A(object a) { }

	/// <inheritdoc cref="A(object)"/>
	/// <param name="b">hello world</param>
	public static void B(object a, object b) { } // <- CS1573 on "a"
}

/// <summary>hello world</summary>
public static class CS1712_Issue
{
	/// <summary>hello world</summary>
	/// <typeparam name="Ta">hello world</typeparam>
	public static void A<Ta>() { }

	/// <inheritdoc cref="A{Ta}"/>
	/// <typeparam name="Tb">hello world</typeparam>
	public static void B<Ta, Tb>() { } // <- CS1712 on "a"
}
```

@sharwell

**Comment by saucecontrol** [comment]
Primary constructors are another case where inheriting partial `param` docs would be useful.  This is now triggering CS1573:

![image](https://github.com/dotnet/roslyn/assets/7929102/4f1d459f-a522-4b3a-939b-6740eb4524e6)

**Comment by smaillet** [comment]
While the finger pointing debate on who owns this problem rages on, can this at least be turned into a warning we can use the SuppressMessageAttribute on? At least that way we can localize disabling it to the impacted functions and not need to clutter ugly pragmas everywhere or at file scope. (Use of the attribute is still intrusive and ugly but is at least a step better than the pragmas.

**Comment by Blackclaws** [comment]
I just looked into this and I'm astounded this has been going since 2019, can we get any statement from the involved teams on what the state of this is and whether its even planned to be addressed at any point?
